### PR TITLE
link from project page to enter expenses and return redirections

### LINKED ISSUE #354: Worker Redirect to homepage after filling in expense and add confirmation alert

### PR DESCRIPTION
**Pull request by srkirkland**
closes #354

### CODE REVIEW

File: Harvest.Web/ClientApp/src/Projects/ProjectDetailContainer.tsx [hunk 1]
```diff
@@ -78,6 +78,17 @@ export const ProjectDetailContainer = () => {
         <div className="card-content">
           <div className="row justify-content-between">
             <div className="col">
+              <ShowFor
+                roles={["Worker", "Supervisor", "FieldManager"]}
```
Comment (jSylvestre) [inline Harvest.Web/ClientApp/src/Projects/ProjectDetailContainer.tsx:82]: @srkirkland I don't think workers can see the list of projects to get here.

Comment (srkirkland) [reply Harvest.Web/ClientApp/src/Projects/ProjectDetailContainer.tsx:82]: Good point

File: Harvest.Web/ClientApp/src/Expenses/ExpenseEntryContainer.tsx [hunk 2]
```diff
@@ -98,7 +101,12 @@ export const ExpenseEntryContainer = () => {
     const response = await request;
 
     if (response.ok) {
-      history.push("/project");
+      // go to the project page unless you are a worker -- worker can't see the project page
+      if (roles.includes("Worker")) {
+        history.push("/");
```
Comment (jSylvestre) [inline Harvest.Web/ClientApp/src/Expenses/ExpenseEntryContainer.tsx:106]: @srkirkland This goes to a react home?

image

Comment (srkirkland) [reply Harvest.Web/ClientApp/src/Expenses/ExpenseEntryContainer.tsx:106]: I didn't actually try it but why wouldn't it work?  what's the url?

Comment (jSylvestre) [reply Harvest.Web/ClientApp/src/Expenses/ExpenseEntryContainer.tsx:106]: https://localhost:44308/

Comment (jSylvestre) [reply Harvest.Web/ClientApp/src/Expenses/ExpenseEntryContainer.tsx:106]: If I refresh, the normal home page shows up.

Comment (srkirkland) [reply Harvest.Web/ClientApp/src/Expenses/ExpenseEntryContainer.tsx:106]: Ok i think i see it, we left in some debug code when doing the home route.  it'll take a look and see if that's an easy fix.

Comment (srkirkland) [reply Harvest.Web/ClientApp/src/Expenses/ExpenseEntryContainer.tsx:106]: Yea, we don't actually have a react homepage -- I'll make an issue for that and do it there.

**Review by jSylvestre — APPROVED** [review]

**Pull request merged**